include/llvm-dialects/TableGen/Operations.h

```diff
@@ -141,6 +165,13 @@ class BuilderMethod {
   struct Arg {
     std::string cppType;
     std::string name;
+    std::optional<std::string_view> defaultValue;
+
+    void emitArg(llvm::raw_ostream &os) const {
+      os << cppType << " " << name;
+      if (defaultValue.has_value())
+        os << " = " << defaultValue.value();
+    }
```

Comment:
I don't think we should be adding these default values by default just because it's varargs and there's a simple default to give.

I'm not opposed to the idea of defaults for builder arguments, but it should be something that is chosen explicitly in the .td file. (In particular, there could also be a default for Attr* arguments.)

Reply:
The idea is not to choose an explicit default in general, but to avoid having the user to specify the variadic argument in create calls in case he doesn't need them. So, for varargs, we always want to default to `{}`.
Separating the `Arg` and `BuilderArg` however makes it more complex to emit a default value based on the kind of the argument.
When going with the approach of defining the argument defaults in the dialect, we could create a custom DAG node for varargs where {} always is the default, but is there really any use case for defining other default values at this stage?

Reply:
If we can live with having the caller pass an empty array, that'd be fine from my side, and I'll remove the defaults again.